src/init/starship.fish

```diff
@@ -43,8 +44,9 @@ function fish_right_prompt
             printf ""
         end
     else
-        ::STARSHIP:: prompt --right --terminal-width="$COLUMNS" --status=$STARSHIP_CMD_STATUS --pipestatus="$STARSHIP_CMD_PIPESTATUS" --keymap=$STARSHIP_KEYMAP --cmd-duration=$STARSHIP_DURATION --jobs=$STARSHIP_JOBS
+        ::STARSHIP:: prompt $STARSHIP_ADD_NEWLINE_FLAGS --right --terminal-width="$COLUMNS" --status=$STARSHIP_CMD_STATUS --pipestatus="$STARSHIP_CMD_PIPESTATUS" --keymap=$STARSHIP_KEYMAP --cmd-duration=$STARSHIP_DURATION --jobs=$STARSHIP_JOBS
     end
+    set -g STARSHIP_ADD_NEWLINE_FLAGS
```

Comment:
This approach is incompatible with vi-mode (I assume the same goes for pwsh).

Reply:
😢 Thanks for finding this, I agree it does not work for fish
Reproduction is `fish_vi_key_bindings`, clearing the screen, and then pressing escape.
Unfortunately this clear hook approach doesn't really work then.

I tested the same with powershell using `Set-PSReadlineOption -EditMode vi` and it works fine, since it always retrieves the cursor's position 👍

Reply:
Just flipping between vi modes causes an extra line to appear for me, even without clearing. Pwsh might only be working for you thanks to the y-position check.

Reply:
Yeah, and the y-position check could work for fish and perhaps other shells (won't work for zsh/bash on clear since the prompt is not re-evaluated on clear, iirc).

The way pwsh on mac queries the position [seems quite involved](https://github.com/dotnet/runtime/blob/8f5364695b107f37af899aba3ad1b9f98d577c33/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L437-L672), it might be an idea to query the cursor position from rust on nix based machines in a similar way.

Cmd would probably use win32 the same as pwsh/powershell on windows I think.
Many ideas 🤔